Criterion/Measurement.hs

```diff
@@ -65,14 +65,23 @@ measure (Benchmarkable run) iters = do
   endCycles <- getCycles
   endEnergy <- getEnergy
   endStats <- getGCStats
-  let !m = applyGCStats endStats startStats $ measured {
-             measTime    = max 0 (endTime - startTime)
-           , measCpuTime = max 0 (endCpuTime - startCpuTime)
-           , measCycles  = max 0 (fromIntegral (endCycles - startCycles))
-           , measEnergy  = max 0 (endEnergy - startEnergy)
-           , measIters   = iters
-           }
-  return (m, endTime)
+
+  if endEnergy < startEnergy
+    then do
+      let st = show startEnergy
+          ed = show endEnergy
+          it = show iters
+      putStrLn $ "OVERFLOW DETECTED! start: " ++ st ++ ". end: " ++ ed ++ ". niters: " ++ it
+      measure b iters
```

Comment:
If the benchmark respects the precondition described in the commit message (the execution time  < full overflow period) I don't think it can lead to an infinity loop.

But we're not enforcing this precondition in the code, so yes, you're right.